src/pcap/pcap.go

```diff
@@ -191,3 +193,52 @@ func vcapIDFromCtx(ctx context.Context) (*string, error) {
 
 	return nil, errNoVcapId
 }
+
+var interfaceAddrs = net.InterfaceAddrs
+
+// patchFilter extends the given filter by excluding the filter generated
+// by generateApiFilter.
+func patchFilter(filter string) (string, error) {
+	apiFilter, err := generateApiFilter()
+	if err != nil {
+		return "", err
+	}
+
+	if filter == "" {
+		return fmt.Sprintf("not (%s)", apiFilter), nil
+	} else {
+		return fmt.Sprintf("not (%s) and (%s)", apiFilter, filter), nil
```

Comment:
You could technically do the same thing with the `join` here as you do below.

Reply:
No.